sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -223,6 +223,9 @@ object JdbcUtils extends Logging {
       case java.sql.Types.STRUCT        => StringType
       case java.sql.Types.TIME          => TimestampType
       case java.sql.Types.TIMESTAMP     => TimestampType
+      case java.sql.Types.TIMESTAMP_WITH_TIMEZONE
+                                        => TimestampType
+      case -101                         => TimestampType
```

Comment:
I did a try. The fix works in my local docker test environment. Below is the way how we run docker integration test. 

```
build/mvn -Pyarn -Phadoop-2.6 -Dhadoop.version=2.6.0  -Phive-thriftserver -Phive -DskipTests  install

Before running dockers test . You need to set the docker env variabled. Following does the magic:
eval $(docker-machine env default)

build/mvn -Pdocker-integration-tests -pl :spark-docker-integration-tests_2.11  compile test
```

Feel free to ping me if you hit any issue.